win32ss/printing/base/winspool/printers.c

```diff
@@ -1031,16 +1031,469 @@ GetDefaultPrinterW(LPWSTR pszBuffer, LPDWORD pcchBuffer)
+                    ERR("HeapAlloc failed!\n");
+                    goto Cleanup;
+                }
+
+                WideCharToMultiByte(CP_ACP, 0, ppi4w->pServerName, -1, pszServerName, cch + 1, NULL, NULL);
+                StringCchCopyA(ppi4a->pServerName, cch + 1, pszServerName);
+
+                HeapFree(hProcessHeap, 0, pszServerName);
+            }
+        break;
+        }
+
+        case 5:
+        {
+            if (ppi5w->pPrinterName)
+            {
+                PSTR pszPrinterName;
+
+                // Convert Unicode pPrinterName to a ANSI string pszPrinterName.
+                cch = wcslen(ppi5w->pPrinterName);
+
+                pszPrinterName = HeapAlloc(hProcessHeap, 0, (cch + 1) * sizeof(CHAR));
+                if (!pszPrinterName)
+                {
+                    SetLastError(ERROR_NOT_ENOUGH_MEMORY);
+                    ERR("HeapAlloc failed!\n");
+                    goto Cleanup;
+                }
+
+                WideCharToMultiByte(CP_ACP, 0, ppi5w->pPrinterName, -1, pszPrinterName, cch + 1, NULL, NULL);
+                StringCchCopyA(ppi5a->pPrinterName, cch + 1, pszPrinterName);
+
+                HeapFree(hProcessHeap, 0, pszPrinterName);
+            }
+
+            if (ppi5w->pPortName)
+            {
+                PSTR pszPortName;
+
+                // Convert Unicode pPortName to a ANSI string pszPortName.
+                cch = wcslen(ppi5w->pPortName);
+
+                pszPortName = HeapAlloc(hProcessHeap, 0, (cch + 1) * sizeof(CHAR));
+                if (!pszPortName)
+                {
+                    SetLastError(ERROR_NOT_ENOUGH_MEMORY);
+                    ERR("HeapAlloc failed!\n");
+                    goto Cleanup;
+                }
+
+                WideCharToMultiByte(CP_ACP, 0, ppi5w->pPortName, -1, pszPortName, cch + 1, NULL, NULL);
+                StringCchCopyA(ppi5a->pPortName, cch + 1, pszPortName);
+
+                HeapFree(hProcessHeap, 0, pszPortName);
+            }
+        break;
+        }
+
+        case 7:
+        {
+            if (ppi7w->pszObjectGUID)
+            {
+                PSTR pszaObjectGUID;
+
+                // Convert Unicode pszObjectGUID to a ANSI string pszaObjectGUID.
+                cch = wcslen(ppi7w->pszObjectGUID);
+
+                pszaObjectGUID = HeapAlloc(hProcessHeap, 0, (cch + 1) * sizeof(CHAR));
+                if (!pszaObjectGUID)
+                {
+                    SetLastError(ERROR_NOT_ENOUGH_MEMORY);
+                    ERR("HeapAlloc failed!\n");
+                    goto Cleanup;
+                }
+
+                WideCharToMultiByte(CP_ACP, 0, ppi7w->pszObjectGUID, -1, pszaObjectGUID, cch + 1, NULL, NULL);
+                StringCchCopyA(ppi7a->pszObjectGUID, cch + 1, pszaObjectGUID);
+
+                HeapFree(hProcessHeap, 0, pszaObjectGUID);
+            }
+        }
+    }       // switch
+
+Cleanup:
+
+    return bReturnValue;
```

Comment:
This doesn't seem like a cleanup body since it just returns the value by the function?

Reply:
True. This was done to be consistent with the other functions in this program and provides a common return point. It is also the same as the code for EnumPrintersA. Do you have a specific suggestion for a change? Thanks.